demisto_sdk/commands/update_release_notes/update_rn_manager.py

```diff
@@ -0,0 +1,194 @@
+                                               silence_init_prints=True, skip_conf_json=True, check_is_unskipped=False)
+            if not validate_manager.git_util:  # in case git utils can't be initialized.
+                raise git.InvalidGitRepositoryError('git_utils is not initialized.')
+            validate_manager.setup_git_params()
+            return validate_manager.get_changed_files_from_git()
+        except (git.InvalidGitRepositoryError, git.NoSuchPathError, FileNotFoundError) as e:
+            raise FileNotFoundError(
+                "You are not running `demisto-sdk update-release-notes` command in the content repository.\n"
+                "Please run `cd content` from your terminal and run the command again") from e
+
+    def check_existing_rn(self, added_files: set):
+        """
+            Checks whether the packs already have an existing release notes files and adds
+            them to the packs_existing_rn dictionary.
+
+            :param added_files: A set of new added files
+        """
+        for file_path in added_files:
+            if 'ReleaseNotes' in file_path:
+                self.packs_existing_rn[get_pack_name(file_path)] = file_path
+
+    def handle_api_module_change(self, modified_files: set, added_files: set):
+        """ Checks whether the modified file is in the API modules pack and if so, updates every pack which depends
+            on that API module.
+
+            :param
+                added_files: A set of new added files
+                modified_files: A set of modified files
+        """
+        # The user gave a path to the api module which was changed or he didn't give a path but some api modules
+        # have changed.
+        api_module_was_given = self.given_pack and API_MODULES_PACK in self.given_pack
+        api_module_changed_in_git = self.changed_packs_from_git and API_MODULES_PACK in self.changed_packs_from_git
+
+        if api_module_was_given or api_module_changed_in_git:
+            updated_packs = update_api_modules_dependents_rn(self.pre_release, self.update_type, added_files,
+                                                             modified_files, self.id_set_path, self.text)
+            self.total_updated_packs = self.total_updated_packs.union(updated_packs)
+
+    def create_release_notes(self, modified_files: set, added_files: set, old_format_files: set):
+        """ Iterates over the packs which needs an update and creates a release notes for them.
+
+            :param
+                modified_files: A set of modified files
+                added_files: A set of added files
+                old_format_files: A set of old formatted files
+        """
+        # Certain file types do not require release notes update
+        filtered_modified_files = filter_files_by_type(modified_files, skip_file_types=SKIP_RELEASE_NOTES_FOR_TYPES)
+        filtered_added_files = filter_files_by_type(added_files, skip_file_types=SKIP_RELEASE_NOTES_FOR_TYPES)
```

Comment:
I think the same functionality exists in validate manager in the rn validator method.
If so it could be wise to make a tool for this in order to keep the consistency between the commands.
Another possibility is to keep the functionality in validate_manager and import it here in the same fashion you do for git functions.

Reply:
"filter_files_by_type" is imported from tools. If you are talking about "filter_to_relevant_files" function in validate_manager, then I think they are not quite the same. It seems that "filter_to_relevant_files" is more specific for the validation process while the filter_files_by_type is more general and simple.